### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by elhoyos**

### CODE REVIEW

File: lib/proxy.js [hunk 1]
```diff
@@ -51,13 +50,13 @@ function HttpProxy (options) {
...
-    self.proxyRequest(req, res);
+  this.server = http.createServer((req, res) => {
+    this.proxyRequest(req, res);
   });
 
   //Support for HTTP Tunneling: http://en.wikipedia.org/wiki/HTTP_tunnel
-  this.server.on("connect", function (req, clientSocket, head) {
-    self.tunelRequest(req, clientSocket, head);
+  this.server.on("connect", (req, clientSocket, head) => {
+    thiself.tunelRequest(req, clientSocket, head);
```
Comment (danypype) [inline lib/proxy.js:59]: Typo here

File: test/test.js [hunk 2]
```diff
@@ -111,7 +107,7 @@ describe("HttpProxy", function () {
       })
     });
 
-    describe("Concurrency", function () {
+    describe.only("Concurrency", function () {
```
Comment (danypype) [inline test/test.js:110]: Please remove `.only`

File: test/test.js [hunk 3]
```diff
@@ -136,26 +132,31 @@ describe("HttpProxy", function () {
...
-      it("Handles concurrency of http requests", function (done) {
-        testCaseForConcurrency("http", this.response, this.n, done);
+      it("Handles concurrency of http requests", function () {
+        return testCaseForConcurrency("http", this.response, this.n);
       });
 
-      it("Handles concurrency of https requests", function (done) {
-        testCaseForConcurrency("https", this.response, this.n, done);
-      });
+      // it("Handles concurrency of https requests", function () {
```
Comment (danypype) [inline test/test.js:157]: Should we enable this test case?

**Review by danypype — CHANGES_REQUESTED** [review]
Thank you, @elhoyos. I left some comments inline with the code.

**Comment by elhoyos** [comment]
Dude, this is a WIP, sorry for not telling you later. A PR is unnecessary, just run the tests in the branch.

**Pull request closed without merging**